app/AttributionGenerator.js

```diff
@@ -158,7 +158,9 @@ $.extend( AttributionGenerator.prototype, {
 		// Prefer attribution over author:
 		var $attribution = this._asset.getAttribution();
 		if( $attribution ) {
-			return format === 'html' ? $attribution : $attribution.text();
+			if( $attribution.length && $attribution.text().length ) {
```

Comment:
In JavaScript this is common, we also do this in Wikibase related JavaScript code.